server/service/validator.js

```diff
@@ -0,0 +1,66 @@
+const {BadRequestError} = require('../errorTypes/errorTypes')
+module.exports = {
+    validateAttributes: (modelAttributes, attributes) => {
+        //filtering attributes which aren't in the schema
+
+        const modelKeys = Object.keys(modelAttributes);
+        const filteredAttributes = Object.keys(attributes)
+            .filter(key => modelKeys.includes(key))
+            .reduce((acc, key) => Object.assign(acc, { [key]: attributes[key] }), {});
+
+
+        //check required
+        modelKeys.forEach(attributeKey => {
+            if (modelAttributes[attributeKey].required && filteredAttributes[attributeKey] == undefined) {
```

Comment:
better to change filteredAttributes[attributeKey] == undefined 
to !filteredAttributes[attributeKey]

Reply:
Done